merge branches

shanyu-e opened the pull request.

ghost commented:
## DeepCode's analysis on [#63fd97](https://www.deepcode.ai/app/gh/xiaol/you-get/a87fc22e847ef8f476fa8f450badf011b6fdfcd5/xiaol/you-get/63fd9716a8740fc6862b70a474e398ca6e9f26bd/pr/_/%2F/code/?utm_source=gh_review&c=3&w=6&i=5&) found:

- :x: **3** critical issues, :warning: **6** warnings, :information_source: **5** minor issues. :point_down:
- :heavy_check_mark: **5** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Do not use old versions of TLS (ssl.PROTOCOL_TLSv1 used in ssl.SSLContext). Occurrences: <ul><li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractors/infoq.py#L33">infoq.py:33</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/xiaol/you-get/a87fc22e847ef8f476fa8f450badf011b6fdfcd5/xiaol/you-get/63fd9716a8740fc6862b70a474e398ca6e9f26bd/pr/_/%2Fsrc%2Fyou_get%2Fextractors%2Finfoq.py/python%2Fdc_interfile_project%2FDoNotUseOldTls/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Use binary mode in open (current mode is w) to avoid encoding-related issues for written file, on Windows or with Python 3. Occurrences: <ul><li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractors/iqiyi.py#L215">iqiyi.py:215</a></li> <li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractor.py#L251">extractor.py:251</a></li> <li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractor.py#L259">extractor.py:259</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/xiaol/you-get/a87fc22e847ef8f476fa8f450badf011b6fdfcd5/xiaol/you-get/63fd9716a8740fc6862b70a474e398ca6e9f26bd/pr/_/%2Fsrc%2Fyou_get%2Fextractors%2Fiqiyi.py/python%2Fdc_interfile_project%2FBinaryWrite/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Avoid hardcoding values that are meant to be secret. Found a hardcoded string used in here. Occurrences: <ul><li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractors/vimeo.py#L12">vimeo.py:12</a></li> <li><a href="https://github.com/xiaol/you-get/blob/63fd9716a8740fc6862b70a474e398ca6e9f26bd/src/you_get/extractors/vimeo.py#L24">vimeo.py:24</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/xiaol/you-get/a87fc22e847ef8f476fa8f450badf011b6fdfcd5/xiaol/you-get/63fd9716a8740fc6862b70a474e398ca6e9f26bd/pr/_/%2Fsrc%2Fyou_get%2Fextractors%2Fvimeo.py/python%2Fdc_interfile_project%2FHardcodedNonCryptoSecret/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/xiaol/you-get/a87fc22e847ef8f476fa8f450badf011b6fdfcd5/xiaol/you-get/63fd9716a8740fc6862b70a474e398ca6e9f26bd/pr/_/%2F/code/?utm_source=gh_review&c=3&w=6&i=5&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=xiaol)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

soimort committed:
[soundcloud] fix

soimort committed:
[youtube] partial fix for #2857

soimort committed:
[youtube] fully fix #2857 with DASH stream

soimort committed:
Merge branch 'patch-1' of https://github.com/SuperHuangXu/you-get into SuperHuangXu-patch-1

soimort committed:
[yinyuetai] purge (#2855)

soimort committed:
version 0.4.1500

snowyu committed:
fix(lrts): can not download audio for the count less than pagesize

soimort committed:
Merge branch 'feat/lrts' of https://github.com/snowyu/you-get into snowyu-feat/lrts

soimort committed:
[common] use only double-dashed arguments

sinofool committed:
Fix mgtv: adds tk2 and referer

sinofool committed:
Mgtv: add another format of url
Fix indentation causing undefined variable

otomn committed:
ignore vim swap files

soimort committed:
[youtube] fix extraction of caption tracks, close #2866

soimort committed:
[xiami] xiami is dead

soimort committed:
[tumblr] fix extraction

soimort committed:
[instagram] data not none even when missing cookies

v4hn committed:
[youtube] fix playlist extraction & comprehensive error output

soimort committed:
Merge branch 'develop' of https://github.com/const-zhou/you-get into const-zhou-develop

SHMoney2021 committed:
fix page count error in 'space_video' case

ljhcage committed:
add HDR support for bilibili

soimort committed:
[test] remove test_bilibili (videos deleted)

soimort committed:
version 0.4.1520

soimort committed:
[bilibili] fix a bug for non-interative multi-part videos (https://github.com/soimort/you-get/pull/2746#pullrequestreview-626492105)

soimort committed:
[bilibili] redirect /s/ URLs

flewsea committed:
skip private video

bryanzhou committed:
add param m3u8 , allow download video via m3u8 url

soimort committed:
[youtube] close #2890

soimort committed:
test on Python 3.9

soimort committed:
version 0.4.1525

soimort committed:
[youtube] s_to_sig: add prefix to prevent namespace pollution

soimort committed:
version 0.4.1527

nuomi1 committed:
fix: url

nuomi1 committed:
feat: new missevan_stream_types

pyx committed:
Update douyin.py

The site douyin.com changed to a more user friendly website, this is the updated extractor for that.

zhancat200801 committed:
modify qq.py

pyx committed:
Update douyin.py

Updated the extractors for the latest www.douyin.com

Kagamia committed:
miaopai: use weibo h5 api

soimort committed:
[common] manually add cookies in get_content and post_content (due to python's lack of support for #HttpOnly_)

soimort committed:
[instagram] use get_content

soimort committed:
Merge branch '20210711-fix-miaopai' of https://github.com/Kagamia/you-get into Kagamia-20210711-fix-miaopai

soimort committed:
version 0.4.1536

timgates42 committed:
docs: Fix a few typos

There are small typos in:
- src/you_get/extractors/flickr.py
- src/you_get/extractors/mtv81.py
- src/you_get/extractors/qingting.py

Fixes:
- Should read `several` rather than `serveral`.
- Should read `channel` rather than `chaanel`.
- Should read `approach` rather than `approch`.

soimort committed:
[youtube] tr_js: support 3-char main function names

soimort committed:
Merge branch 'master' of https://github.com/flewsea/you-get into flewsea-master

soimort committed:
Merge branch 'flewsea-master' into develop

soimort committed:
[test] disable test_soundcloud temporarily

soimort committed:
Merge branch 'bugfix_typos' of https://github.com/timgates42/you-get into timgates42-bugfix_typos

soimort committed:
[test] disable test_soundcloud temporarily

soimort committed:
version 0.4.1545

Commit:
fix for using insecure flag does not work

soimort committed:
[universal] fix blogger

soimort committed:
Merge branch 'you-get-fs' of https://github.com/zhancat200801/you-get into zhancat200801-you-get-fs

sartoshi-foot-dao committed:
Create SECURITY.md

I'd like to report a security issue but cannot find contact instructions on your repository.

If not a hassle, might you kindly add a SECURITY.md file with an email, or another contact method? GitHub [recommends](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) this best practice to ensure security issues are responsibly disclosed, and it would serve as a simple instruction for security researchers in the future.

Thank you for your consideration!

wantee committed:
support channel/series for bilibili

soimort committed:
[universal] tweak

cclauss committed:
Test on Python 3.10

“3.10” must be quoted or yaml will treat it as 3.1.

cclauss committed:
strategy: fail-fast: false

cclauss committed:
"Programming Language :: Python :: 3.10",

chuangzhu committed:
[bilibili] fix 'NoneType' object is not subscriptable

soimort committed:
Merge branch 'patch-1' of https://github.com/cclauss/you-get into cclauss-patch-1

soimort committed:
version 0.4.1555

soimort committed:
[test] reduce test_youtube

songzy12 committed:
Fix zhihu extractor comment: ld

soimort committed:
Merge branch 'bilibili-series' of https://github.com/wantee/you-get into wantee-bilibili-series

soimort committed:
Merge branch 'develop' of https://github.com/bryanzhou/you-get into bryanzhou-develop

soimort committed:
fix the "name m3u8 is not defined" bug of #2888

soimort committed:
add param "--postfix" to postfix downloaded files with unique identifiers

soimort committed:
[imgur] fix extraction of original image

soimort committed:
[imgur] prefer og:video

peter279k committed:
Add pipenv command usage option

soimort committed:
[instagram] fix extraction

soimort committed:
[instagram] fix extraction

soimort committed:
[instagram] fix video extraction

SuperSodaSea committed:
[bilibili] Add 8K video download support

SuperSodaSea committed:
[bilibili] Add Dolby Vision video download support

sdlyyxy committed:
Bilibili: fix "真彩 HDR" video_resolution

fakeboboliu committed:
change -safe to 0 for ffmpeg5

https://git.ffmpeg.org/gitweb/ffmpeg.git/commit/46fb395952be32692385449ee214461603480b4b

ljhcage committed:
update space_video_api arg: page size

soimort committed:
[instagram] fix

licolas94 committed:
修复西瓜视频下载失败问题;
fix problem:the JSON object must be str, bytes or bytearray, not NoneType

licolas94 committed:
优化ixigua.py代码

licolas94 committed:
优化ixigua.py代码

ljhcage committed:
clear stream info

soimort committed:
[youtube] s_to_sig: no substitution for 3-char main function names

soimort committed:
[youtube] support shorts

soimort committed:
[common] implement getHttps

soimort committed:
[tests] fix test_bilibili

soimort committed:
[tests] disable test_bilibili (works for me, dunno why)

soimort committed:
[tests] enable test_tiktok

soimort committed:
[tiktok] fix extraction

soimort committed:
Merge branch 'patch-2' of https://github.com/ljhcage/you-get into ljhcage-patch-2

soimort committed:
Merge branch 'sdlyyxy' of https://github.com/sdlyyxy/you-get into sdlyyxy-sdlyyxy

soimort committed:
Merge branch 'develop' of https://github.com/SuperSodaSea/you-get into SuperSodaSea-develop

soimort committed:
[bilibili] use hdflv2_hdr as id for HDR

soimort committed:
Merge branch 'develop' of https://github.com/albertli2333/you-get into albertli2333-develop

soimort committed:
Merge branch 'add_pipfile_support' of https://github.com/peter279k/you-get into peter279k-add_pipfile_support

soimort committed:
Merge branch 'develop' of https://github.com/hong56hk/you-get into hong56hk-develop

soimort committed:
drop support for python <3.5

soimort committed:
[bilibili] support lowercase bv

soimort committed:
[twitter] show the warning message if login required

soimort committed:
[tiktok] fix extraction

soimort committed:
[bilibili] support festival videos (fix #2955)

soimort committed:
[bilibili] oops

soimort committed:
[common] fix google_search (close #2924)

soimort committed:
version 0.4.1602

soimort committed:
[tiktok] improve regexp

yummydish committed:
Fixed a typo

soimort committed:
[common] new parameters for getHttps

soimort committed:
[tiktok] fix extraction for alternative URLs

soimort committed:
[tests] test_twitter

soimort committed:
[youtube] improve regex

soimort committed:
[.github/workflows] test pypy3.8 and pypy3.9

soimort committed:
[.github/workflows] test pypy-3.8 and pypy-3.9

soimort committed:
drop support for python <3.7.4

soimort committed:
version 0.4.1612

soimort committed:
update README.md

soimort committed:
update README.rst

soimort committed:
update you-get.json

soimort committed:
update Makefile

soimort committed:
[youtube] we should extract ytInitialPlayerResponse more reliably

soimort committed:
[twitter] better warning

soimort committed:
[instagram] fix extraction

soimort committed:
version 0.4.1620

owlwang committed:
fix douyin extractor

soimort committed:
[youtube] fix s_to_sig

soimort committed:
[tiktok] fix extraction for alternative URLs

foool committed:
Subtitute the ffmpeg download method.

Subtitute the ffmpeg download method.

foool committed:
narrow the search of tvid and videoid

test script `python3 you-get https://www.iqiyi.com/a_19rrha613h.html -l --debug`

interiv committed:
Update youku.py

ccode 0532 to 0564

cclauss committed:
Add Python 3.11 release candidate 2 to the testing

Python 3.11 will be on average 22% faster than Python 3.10 so let’s give it a spin...
https://www.python.org/download/pre-releases

xiangxw committed:
[ixigua] fix 'string indices must be integers'

StdioA committed:
[netease]: Add pagination on netease radio

0x4ea5 committed:
added support to bilibili collectiondetail api

nuomi1 committed:
fix: use fake ua

nuomi1 committed:
fix: remove query and fragment

soimort committed:
Merge branch 'improve/radio-paginate' of https://github.com/StdioA/you-get into StdioA-improve/radio-paginate

soimort committed:
Merge branch 'fix_ixigua' of https://github.com/xiangxw/you-get into xiangxw-fix_ixigua

soimort committed:
Merge branch 'develop' of https://github.com/0x4ea5/you-get into 0x4ea5-develop

soimort committed:
Merge branch 'patch-1' of https://github.com/cclauss/you-get into cclauss-patch-1

soimort committed:
[twitter] support tweets with multiple videos

soimort committed:
[twitter] small fix

zhiwenzheng committed:
test miaopai

sankoshine committed:
[douyin] remove video watermark

zhiwenzheng committed:
miaopai fix

soimort committed:
Merge branch 'develop' of https://github.com/sankoshine/you-get into sankoshine-develop

soimort committed:
[twitter] support NSFW tweets

soimort committed:
[twitter] if the tweet contains video from another tweet, download it

soimort committed:
[twitter] small fix

soimort committed:
[tiktok] fix extraction

soimort committed:
[tiktok] fix extraction for alternative URLs

soimort committed:
[tests] test "universal" tiktok url

soimort committed:
[instagram] show cookie warning

soimort committed:
version 0.4.1650

soimort committed:
[twitter] warn when falling back to deprecated API

arix00 committed:
Download multipage video collection

When there're more than single page videos in a collection,
Download all videos as current code will only handle first page.

For 'space_channel_series' and 'space_channel_collection'

ByteTuxiaobei committed:
Update ffmpeg.py

URenko committed:
support different codecs for bilibili

soimort committed:
add param "--prefix" to prefix downloaded files

soimort committed:
Merge branch 'develop' of https://github.com/URenko/you-get into URenko-develop

soimort committed:
Merge branch 'arix00-patch-1' of https://github.com/arix00/you-get-test into arix00-arix00-patch-1

soimort committed:
[bilibili] warn if cookies are not loaded

soimort committed:
[.github/workflows] test python 3.11

soimort committed:
[twitter] fix extraction

soimort committed:
[twitter] fix extraction

soimort committed:
[twitter] minor fix

soimort committed:
[common] update UA

linzhanyu committed:
[bilibili] api url update.

johzzy committed:
[twitter] fix twitter video download

ifui committed:
fix ffmpeg: Unrecognized option 'absf'.

ifui committed:
Merge branch 'soimort:develop' into develop

soimort committed:
Merge branch 'develop' of https://github.com/johzzy/you-get into johzzy-develop

soimort committed:
Merge branch 'develop' of https://github.com/ifui/you-get into ifui-develop

soimort committed:
[imgur] fix extraction

soimort committed:
[imgur] fix extraction

soimort committed:
[common] update UA

crnkv committed:
The installation instructions failed for two reasons:
1. python 3.12 has removed the `imp` module and has replaced it by the `importlib` module. [1](https://docs.python.org/3/whatsnew/3.12.html#imp)
2. the use of setup.py as in the command `python setup.py install` is deprecated and should be replaced by the recommended way of `python -m pip install path/to/project`. [2](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/)

ftakanashi committed:
fix: [extractors.bilibili] add headers when requesting for danmaku

soimort committed:
Merge branch 'develop' of https://github.com/ftakanashi/you-get into ftakanashi-develop

soimort committed:
[bilibili] change all http to https

soimort committed:
[twitter] fix extraction (x.com)

crnkv committed:
Update setup.py: compatibility for older python versions

crnkv committed:
Update python-package.yml: add python-version 3.12 job

crnkv committed:
Update you-get.json: add info Python :: 3.11 and 3.12

crnkv committed:
Merge branch 'fix-installation' of https://github.com/crnkv/you-get into fix-installation

crnkv committed:
Update python-package.yml: update setuptools

crnkv committed:
Revert "Update setup.py: compatibility for older python versions"

This reverts commit 873ffdb61eb461b4c71ab7e12151864b49a23f3a.

Minimun python version 3.7 already assures importlib availability

soimort committed:
[tests] disable test_tiktok temporarily

soimort committed:
[imgur] fix extraction

soimort committed:
Merge branch 'fix-installation' of https://github.com/crnkv/you-get into crnkv-fix-installation

soimort committed:
python-package.yml: add pypy-3.10

soimort committed:
[tiktok] fix extraction

soimort committed:
[tests] enable test_tiktok, etc.

soimort committed:
[tests] remove test_acfun (404 URL)

soimort committed:
[tests] update test_acfun

soimort committed:
update README.md (pip3 -> pip)

soimort committed:
[tumblr] print info for each pic

soimort committed:
[tumblr] add prefix

soimort committed:
update README.md (http:// -> https://)

soimort committed:
[vine] remove support (discontinued)

soimort committed:
update LICENSE.txt

soimort committed:
Merge branch 'patch-1' of https://github.com/zidingz/you-get into zidingz-patch-1

soimort committed:
version 0.4.1700

soimort committed:
[tests] disable test_bilibili temporarily

soimort committed:
[youtube] fix 403 error and throttling (close #2950)

soimort committed:
[youtube] remove dead code

soimort committed:
[youtube] check playabilityStatus and ask for cookies if needed

soimort committed:
[youtube] show subreason if possible

soimort committed:
[youtube] ask for cookies only if status is LOGIN_REQUIRED

soimort committed:
[youtube] prompt for installing dukpy if ImportError

soimort committed:
switch from "setup.py test" to "python -m unittest" (pytest has a weird 'I/O operation on closed file' bug though)

soimort committed:
add requirements.txt

soimort committed:
version 0.4.1710

soimort committed:
Makefile: stop using python setup.py commands

chenrui333 committed:
chore: update `MANIFEST.in` to include `contrib`

chenrui333 committed:
chore: also include `you-get.plugin.zsh` and `CONTRIBUTING.md`

soimort committed:
[common] update UA

soimort committed:
[youtube] more detailed logging of stream formats

soimort committed:
[youtube] separate files for autogenerated caption tracks

soimort committed:
[youtube] fix extraction

soimort committed:
version 0.4.1718

soimort committed:
[youtube] sloppy fix

DGideas committed:
Add ytInitialPlayerResponse checker to let user easily know problem

soimort committed:
[instagram] fix extraction

bryanhonof committed:
Add Flox as an installation option

soimort committed:
Merge branch 'bryanhonof-develop' into develop

soimort committed:
[tiktok] support short URLs (vt.tiktok.com)

soimort committed:
[youtube] add self.ua

soimort committed:
[youtube] update self.ua (fix extraction)

soimort committed:
fix "SyntaxWarning: invalid escape sequence" for Python 3.12

soimort committed:
[tests] disable test_youtube temporarily

soimort committed:
[vimeo] cleanup

soimort committed:
version 0.4.1730

doskey committed:
Fix: Restore functionality of -o parameter for Miaopai downloads

The -o parameter, used for specifying output filename, was not functioning correctly when downloading videos from Miaopai. This commit addresses the issue and ensures the parameter works as intended.

ljhcage committed:
remove imported but unused module

soimort committed:
[youtube] fix extraction (close #3031)

soimort committed:
[twitter] fix extraction

soimort committed:
[youtube] fix reason

soimort committed:
Merge branch 'develop' of https://github.com/doskey/you-get into doskey-develop

soimort committed:
Merge branch 'doskey-develop' into develop

soimort committed:
[youtube] fix extraction (player 3bb1f723)

soimort committed:
python-package.yml: update artifact actions

soimort committed:
Merge branch 'develop' of https://github.com/ljhcage/you-get into ljhcage-develop

soimort committed:
[youtube] update self.ua (fix extraction)

soimort committed:
[youtube] fix caption tracks extraction

soimort committed:
version 0.4.1743

soimort committed:
[common] fix a long-standing bug that causes infinite downloading when content-length is missing

soimort committed:
python-package.yml: disable the new flake8 F824 check

soimort committed:
python-package.yml: disable the new flake8 F824 check

soimort committed:
python-package.yml: remove python 3.7 (no longer available in Ubuntu 24.04)

soimort committed:
README.md: add --force-reinstall to pip because it is now necessary for upgrading from a VCS URL to work when the package version is unchanged (https://github.com/pypa/pip/issues/9397, https://github.com/pypa/pip/issues/5780)
